conda/render_recipe.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+"""
+Render recipe:
+- populate dependencies from the pip requirements
+- set the package version,
+- pin spark version.
+"""
+import sys
+
+SPARK_VERSION = 'hail/env/SPARK_VERSION'
+PIP_REQS = 'hail/python/requirements.txt'
+META_TEMPLATE = 'conda/hail/meta-template.yaml'
+META_RENDERED = 'conda/hail/meta.yaml'
+
+if len(sys.argv) <= 1:
+    raise ValueError(
+        'Please, provide package version after as the first positional argument'
+    )
+package_version = sys.argv[1]
+
+with open(SPARK_VERSION, 'r') as file:
+    spark_version = file.read()
+
+dependencies = []
+with open(PIP_REQS, 'r') as f:
+    for line in f:
+        stripped = line.strip()
+        if stripped.startswith('#') or len(stripped) == 0:
+            continue
+        pkg = stripped
+        if pkg.startswith('pyspark'):
+            # already in the template
+            continue
+        # Removing all >, >= and == version pins, that conda-build can't 
+        # always reliably resolve:
+        pkg = pkg.split('>')[0].strip()
+        pkg = pkg.split('==')[0].strip()
+        # conda uses dashes where pip uses underscores
+        pkg = pkg.replace('_', '-')
+        if pkg == 'avro':
+            pkg = 'python-avro'
+        dependencies.append(pkg)
+
+with open(META_TEMPLATE) as inp, open(META_RENDERED, 'w') as out:
```

Comment:
I don't feel as good about doing this line by line, most templating I've seen uses `.replace('{version}', "<version-string>")` on the whole file, or maybe using a formal templating engine?